server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/ContentServiceV1.java

```diff
@@ -0,0 +1,457 @@
+        final String summary = commitMessage.summary();
+        final String detail = commitMessage.detail();
+        final Markup markup = commitMessage.markup();
+
+        final CompletableFuture<Revision> normalizeFuture = repository.normalize(revision);
+        final CompletableFuture<Revision> headFuture = repository.normalize(Revision.HEAD);
+
+        final CompletableFuture<Revision> future = CompletableFutures.combine(
+                normalizeFuture, headFuture, (normalized, head) -> {
+
+                    // If the revision is not same with head revision, just throw an exception here with
+                    // HttpStatus.BAD_REQUEST. If the revision is same with head revision and if
+                    // there's a change while invoking this operation, then will throw an exception with
+                    // HttpStatus.CONFLICT.
+                    if (normalized.compareTo(head) != 0) {
+                        throw newHttpResponseException(HttpStatus.BAD_REQUEST,
+                                                       "must request with the latest revision");
+                    }
+                    return normalized;
+                }).toCompletableFuture();
+
+        return future.thenCompose(normalizedRevision -> execute(
+                Command.push(commitTimeMills, author, repository.parent().name(), repository.name(),
+                             normalizedRevision,
+                             summary, detail, markup, changes)));
+    }
+
+    /**
+     * GET /projects/{projectName}/repos/{repoName}/contents{path}?revision={revision}&
+     * queryType={queryType}&expression={expression}
+     *
+     * <p>Returns the entry of files in the path. This is same with
+     * {@link #listFiles(String, String, String, String)} except that containing the content of the files.
+     * Note that if the {@link HttpHeaderNames#IF_NONE_MATCH} in which has a revision is sent with,
+     * this will await for the time specified in {@link HttpHeaderNames#PREFER}.
+     * During the time if the specified revision becomes different with the latest revision, response back
+     * right away to the client with the {@link WatchResultDto}. {@link HttpStatus#NOT_MODIFIED} otherwise.
+     */
+    @Get("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/contents(?<path>(|/.*))$")
+    public CompletionStage<?> getFiles(@Param("projectName") String projectName,
+                                       @Param("repoName") String repoName,
+                                       @Param("path") String path,
+                                       @Param("revision") @Default("-1") String revision,
+                                       @Param("queryType") Optional<String> queryType,
+                                       @Param("expression") Optional<String> expression,
+                                       @Header("if-none-match") Optional<String> ifNoneMatch,
```

Comment:
Try to use a request converter here, too. :-) I think it'd be better to make an object like `WatchRequest` so that we can move the codes like parsing timeout to the request converter.

Reply:
I realized that we need to make `RequestObject` as optional for this. I'm not sure if it's good idea though. WDYT @hyangtack ?

Reply:
I think we can return `Optional<?>` from a request converter. Not possible?